ranger-server-bundle/src/main/java/io/appform/ranger/server/bundle/resources/RangerResource.java

```diff
@@ -51,9 +51,11 @@ public RangerResource(List<RangerHubClient<T, R>> rangerHubs) {
     @GET
     @Path("/services/v1")
     @Timed
-    public GenericResponse<Set<Service>> getServices() {
+    public GenericResponse<Set<Service>> getServices(
+            @QueryParam("skipReplicationSources") @DefaultValue("false") boolean skipReplicationSources) {
         return GenericResponse.<Set<Service>>builder()
                 .data(rangerHubs.stream()
+                              .filter(hub -> !skipReplicationSources || !hub.isReplicationSource())
```

Comment:
Can we make this an &&, slightly more readable canonically. If !(skipRelicationSources && hub is a replication source)... To a not(a&&b)